### PR TITLE
adds desktop notification filtering

### PR DESCRIPTION
**Pull request by steefmin**
### What

This PR adds filtering to the desktop notification functionality added in #64. 

### Why

Currently, when the desktop notifications are enabled, you will receive a notification of each update of the status of your pipeline. However most users are only interested in two of the cases: 
- the success of your pipeline
and even more important
- a failed pipeline

This PR allows you to select which statuses you want to be notified of. By default the 'error' and 'success' states will be notified, the others are disabled.

### CODE REVIEW

File: front-end/sass/placeholders/_form.sass [hunk 1]
```diff
@@ -29,6 +29,36 @@
...
+
+    &::before
+        content: ' '
+        position: absolute
+        left: 0
+        top: 50%
+        margin-top: -12px
+        width: 20px
+        height: 20px
+        border-radius: 50%
```
Comment (RickvdStaaij) [inline front-end/sass/placeholders/_form.sass]: Please remove the border radius to make it look like an actual checkbox

File: front-end/sass/placeholders/_form.sass [hunk 2]
```diff
@@ -29,6 +29,36 @@
...
+
+    &.current::after
+        content: ' '
+        position: absolute
+        left: 6px
+        top: 50%
+        margin-top: -6px
+        width: 12px
+        height: 12px
+        border-radius: 50%
```
Comment (RickvdStaaij) [inline front-end/sass/placeholders/_form.sass]: Please remove the border radius to make it look like an actual checkbox

File: front-end/components/SettingsPanel/panels/Notify/notify.vue [hunk 3]
```diff
@@ -14,12 +14,23 @@
         <button class="option" @click="enableNotifications" :class="{ current: pushNotifications }">
             Enable desktop notifications
         </button>
+        <ul id="statusSelector">
```
Comment (RickvdStaaij) [inline front-end/components/SettingsPanel/panels/Notify/notify.vue]: what about adding a `v-if="pushNotifications"`? That way the other options are only shown when the notifications are enabled :)

File: front-end/components/SettingsPanel/panels/Notify/notify.vue [hunk 4]
```diff
@@ -48,6 +67,13 @@ export default {
...
 
+.sub-option
+    @extend %check-box
+
 hr
     @extend %line-break
+
+li
+    list-style-type: none
+    padding-left: 20px
```
Comment (RickvdStaaij) [inline front-end/components/SettingsPanel/panels/Notify/notify.vue]: I personally like `35px` better, what do you think?

File: front-end/store/modules/SettingStore.js [hunk 5]
```diff
@@ -5,12 +5,19 @@ import {
...
+    SETTINGS_SET_NOTIFICATION_STATUSES,
 } from '../StaticMutations';
 
 const state = {
     settingsPanelOpen: false,
     theme: 'basic-dark',
     pushNotifications: false,
+    notificationStatuses: {
+        info: false,
+        warning: false,
```
Comment (RickvdStaaij) [inline front-end/store/modules/SettingStore.js]: Can we put everything on `true` by default?

File: front-end/store/modules/SettingStore.js [hunk 6]
```diff
@@ -23,6 +30,10 @@ const actions = {
     [SETTINGS_TOGGLE_NOTIFICATIONS]({ commit, state }) {
         commit(state.pushNotifications ? SETTINGS_SET_NOTIFICATIONS_OFF : SETTINGS_SET_NOTIFICATIONS_ON);
     },
+
+    [SETTINGS_SET_NOTIFICATION_STATUSES]({ commit, state }) {
```
Comment (RickvdStaaij) [inline front-end/store/modules/SettingStore.js]: This action is not needed as it only proxies to the mutation. You can directly call a mutation by `this.$store.commit(SETTINGS_SET_NOTIFICATION_STATUSES, statuses)` in your vue objects. Please remove this one :)

Comment (RickvdStaaij) [reply front-end/store/modules/SettingStore.js]: Ah you're already using `commit`, than this function doesn't do anything at all. Please remove it :)

File: front-end/components/SettingsPanel/panels/Notify/notify.vue (hunk 4 above)
Comment (steefmin) [reply front-end/components/SettingsPanel/panels/Notify/notify.vue]: 35px looks better indeed

**Comment by RickvdStaaij** [comment]
Looks great! Thanks for the improvements @steefmin!

**Pull request merged**